custom_components/frigate/config_flow.py

```diff
@@ -122,7 +127,13 @@ def _show_config_form(
                 {
                     vol.Required(
                         CONF_URL, default=user_input.get(CONF_URL, DEFAULT_HOST)
-                    ): str
+                    ): str,
+                    vol.Optional(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
```

Comment:
As above, for username and password, 'None' might be a better default than an empty string, since it is arguably what everyone has today.

Reply:
i.e. just use `user_input.get(CONF_USERNAME)`